src/gwt/src/org/rstudio/core/client/MouseTracker.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * MouseTracker.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.core.client;
+
+import com.google.gwt.dom.client.NativeEvent;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.Event.NativePreviewEvent;
+import com.google.gwt.user.client.Event.NativePreviewHandler;
+import com.google.inject.Singleton;
+
+@Singleton
+public class MouseTracker
+{
+   public MouseTracker()
+   {
+      Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent preview)
```

Comment:
I'm not sure if there's much to be done / if it's worth it, since IIUC the browser is going to be emitting `mousemove` events regardless, and the amount of work we're doing here is minimal (just caching the mouse coordinates).

FWIW the reason I added this was so that we can know the last mouse coordinates when responding to e.g. keyboard events, in the case where those coordinates are relevant -- this allows us to immediately begin link highlighting after the user initially presses down Ctrl, for example (they don't need to explicitly move the mouse to trigger the click target).